integration_test/tests/power_bi_tests.rs

```diff
@@ -0,0 +1,231 @@
+    }
+
+    /// Test PowerBi connection flow (the setup flow is a pre-requisite)
+    /// Connection flow is :
+    /// - SQLAllocHandle(SQL_HANDLE_DBC)
+    /// - SQLSetConnectAttrW(SQL_ATTR_LOGIN_TIMEOUT)
+    /// - SQLDriverConnectW({NullTerminatedInConnectionString}, SQL_NTS, {NullTerminatedOutConnectionString}, SQL_NTS, SQL_DRIVER_NOPROMPT)
+    /// - SQLGetInfoW(SQL_DRIVER_NAME)
+    /// - SQLGetInfoW(SQL_DBMS_NAME)
+    /// - SQLGetInfoW(SQL_DBMS_VER)
+    #[test]
+    fn test_connection() {
+        let env_handle: HEnv = setup();
+        let (conn_handle, in_connection_string, out_connection_string, output_len) =
+            power_bi_connect(env_handle);
+
+        unsafe {
+            let input_len = in_connection_string.len() as SmallInt;
+
+            println!(
+                "Input connection string = {}\nLength is {}",
+                in_connection_string, input_len
+            );
+            println!(
+                "Output connection string = {}\nLength is {}",
+                out_connection_string, output_len
+            );
+            // The output string should be the same as the input string except with extra curly braces around the driver name
+            assert_eq!(input_len, output_len, "Expect that both connection the input connection string and ouptput connection string have the same length but input string length is {} and output string length is {}",input_len, output_len);
+
+            let str_len_ptr = &mut 0;
+            const BUFFER_LENGTH: SmallInt = 300;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            // SQL_DRIVER_NAME is not accessible through odbc_sys
+            /*
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetInfoW(
+                    dbc as HDbc,
+                    SQL_DRIVER_NAME, // 6
+                    driver_name as Pointer,
+                    BUFFER_LENGTH,
+                    str_len_ptr
+                )
+            );
+             */
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetInfoW(
+                    conn_handle as HDbc,
+                    InfoType::DbmsName,
+                    output_buffer as Pointer,
+                    BUFFER_LENGTH,
+                    str_len_ptr
+                )
+            );
+            println!(
+                "DBMS name = {}\nLength is {}",
+                String::from_utf16_lossy(slice::from_raw_parts(
+                    output_buffer,
+                    *str_len_ptr as usize
+                )),
+                *str_len_ptr
+            );
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetInfoW(
+                    conn_handle as HDbc,
+                    InfoType::DbmsVer,
+                    output_buffer as Pointer,
+                    BUFFER_LENGTH,
+                    str_len_ptr
+                )
+            );
+            println!(
```

Comment:
Do we want to assert anything about these values (DBMS_NAME and VER) instead of just printing them? Or since they could vary (specifically version) are we ok just printing them and manually checking the values are correct?

Reply:
The idea is that we already have/will have unit tests which are validating the values so there is no need to re-check it again.
I print them out to make sure that using the built lib and the DM we get something out and that the buffer length and data match and that in case something looks wrong, we have more trace rather than less to go with when investigating.